src/onnxruntime_benchmark/src/onnxruntime_model.cpp

```diff
@@ -0,0 +1,181 @@
+        // get input name
+        auto input_name = session->GetInputNameAllocated(i, allocator);
+        io.input_names.emplace_back(input_name.get());
+        io.input_names_ptr.push_back(std::move(input_name));
+        // get input type
+        auto type_info = session->GetInputTypeInfo(i);
+        auto tensor_info = type_info.GetTensorTypeAndShapeInfo();
+        ONNXTensorElementDataType type = tensor_info.GetElementType();
+        io.input_data_types.push_back(type);
+
+        // get input shapes/dims
+        auto input_node_shape = tensor_info.GetShape();
+        io.input_shapes.push_back(input_node_shape);
+
+        // log inputs infor
+        logger::info << "\t" << io.input_names[i] << ": " << get_precision_str(get_data_precision(type)) << " "
+                     << shape_string(tensor_info.GetShape()) << logger::endl;
+    }
+
+    // Get outputs from model
+    logger::info << "Model outputs:" << logger::endl;
+    for (size_t i = 0; i < session->GetOutputCount(); ++i) {
+        // get output name
+        auto output_name = session->GetOutputNameAllocated(i, allocator);
+        io.output_names.push_back(output_name.get());
+        io.output_names_ptr.push_back(std::move(output_name));
+
+        // get output type
+        auto type_info = session->GetOutputTypeInfo(i);
+        auto tensor_info = type_info.GetTensorTypeAndShapeInfo();
+        ONNXTensorElementDataType type = tensor_info.GetElementType();
+
+        // log outputs info
+        logger::info << "\t" << io.output_names[i] << ": " << get_precision_str(get_data_precision(type)) << " "
+                     << shape_string(tensor_info.GetShape()) << logger::endl;
+    }
+}
+
+std::vector<ONNXTensorDescr> ONNXModel::get_input_tensors_info() const {
+    std::vector<ONNXTensorDescr> input_tensors_info;
+    for (size_t i = 0; i < io.input_names.size(); ++i) {
+        input_tensors_info.push_back({std::string(io.input_names[i]), io.input_shapes[i], "", io.input_data_types[i]});
+    }
+    return input_tensors_info;
+}
+
+void check_output(const std::vector<Ort::Value> &output_tensors, int batch_size) {
+    for (size_t i = 0; i < output_tensors.size(); ++i) {
+        logger::debug << "Output tensor #" << i << logger::endl;
+        const float *floatarr = output_tensors[i].GetTensorData<float>();
+        for (int b = 0; b < batch_size; ++b) {
+            logger::debug << "Batch #" << b << logger::endl;
+            std::vector<float> res(floatarr + b * 1000, floatarr + b * 1000 + 1000);
+            std::vector<int> idx(1000);
+            std::iota(idx.begin(), idx.end(), 0);
+            std::partial_sort(idx.begin(), idx.begin() + 5, idx.end(), [&res](int l, int r) {
+                return res[l] > res[r];
+            });
+            for (size_t j = 0; j < 5; ++j) {
+                logger::debug << "id: " << idx[j] << " score " << res[idx[j]] << logger::endl;
+            }
+        }
+    }
+}
```

Comment:
Used for testing, will be removed in the future, or placed under conditional compilation and would be enabled only for  debug build.

Reply:
That for classification models only. Consider adding output checking for other types?